src/services/infra/InfraService.ts

```diff
@@ -138,24 +156,17 @@ export default class InfraService {
     const recordsInfo = record.split(" ")
 
     // type checking
-    const schema = Joi.string().domain()
     const sourceUrl = this.removeTrailingDot(recordsInfo[0])
-    if (
-      schema.validate(sourceUrl).error &&
-      // joi reports initial "_" for certificates as as an invalid url WRONGLY,
-      // therefore check if after removing it, it reports as a valid url
-      schema.validate(sourceUrl.substring(1))
-    ) {
+
+    // for the root domain record, Amplify records this as : ' CNAME gibberish.cloudfront.net'.
+    // while this is not a valid URL, this is ok, as it is just an interim representation from Amplify.
+    const isSourceUrlEmpty = sourceUrl === ""
+    if (!isSourceUrlEmpty && !this.isValidUrl(sourceUrl)) {
```

Comment:
this is interesting - does `this.isValidateUrl` return `true` if the url is an empty string? if so, should we strengthen the `isValidUrl` method?

Reply:
hmm i rather not change `isValidUrl()`, since technically it is not a valid url, and there are other callers that depend on this guarantee that an empty string should return `false`

Reply:
i'm not entirely sure i follow here - if an empty string is **not** a valid url, then `!this.isValidUrl(sourceUrl)` would be `!this.isValidUrl("")` which is `!false`, which is `true`. this means that the check for `!isSourceUrlEmpty` would be extra?

conversely, if an empty string **is** a valid url, then we should add an extra check to `isValidUrl` so that the empty string doesn't pass the check

Reply:
> this means that the check for !isSourceUrlEmpty would be extra?

Nope! take the example of `sourceUrl = 'blah'`  which is not an empty string and is not a valid URL (this is unexpected and it should throw an error), and in this case `(!isSourceUrlEmpty && !this.isValidUrl(sourceUrl))` evaluates to `true` and the error is thrown.

conversely, if `sourceUrl = ''` (which is expected, should not throw error), then `(!isSourceUrlEmpty && !this.isValidUrl(sourceUrl)` evaluates to `false`, thus error is not thrown. 

I have changed the code abit in 88ac18c, although there is no change semantically, hopefully this is clearer!

Reply:
Actually my main concern is `this.isValidUrl("")` returning `true`, which seems surprising to me. Why not just have an empty string check inside `isValidUrl`?

Reply:
arh, sorry, I wasnt clear in my first comment :( 

`this.isValidUrl("")` returns `false`!